### PR TITLE
Usecase interface

### PR DESCRIPTION
**Pull request by mariushegele**
Usecases should implement a certain Interface:

`advance` and `is_finished`

They should also return a `Reply` object. 
This policy is enforced by the Controller.

### CODE REVIEW

File: usecase/test/test_work_session.py [hunk 1]
```diff
@@ -31,6 +31,15 @@ def setUp(self):
         usecase.reset()
         self.usecase = usecase
 
+    def test_usecase_is_usecase(self):
```
Comment (Ovakefali13) [inline usecase/test/test_work_session.py]: Nice, method name...

Comment (mariushegele) [reply usecase/test/test_work_session.py]: If you like it so much, you could add that one to your own as well ;)

**Comment by mariushegele** [comment]
> Can you give us an implementation of this advance method ?

For an example of `advance` you can look at #63 and the `work_session` branch

File: usecase/test/test_work_session.py (hunk 1 above)
Comment (Ovakefali13) [reply usecase/test/test_work_session.py]: I guess, but shouldn't you better call it `test_usecase_is_instance`?

**Pull request merged**